lambdaclass/elixir_riak_core
# Elixir VNode

## Changes:
- Working VNode implementation in Elixir.
- Change riax.ex to use the implemented VNode instead of using the Riak's Tutorial VNodes via rpc.
- Change application.ex to start a local node.
- Added VNode Master implementation via riax_sup.ex file (check [here](https://efcasado.github.io/riak-core_intro) why its needed)
- Deleted node{1,2,3}.exs files as they were redundant having the dev{1,2,3}.exs configuration for each node.
- The coverage FSM is still missing, maybe it should be done on another PR.
## How to test:
- Start the nodes in separate terminals with `make node1`,`make node2`, and `make node3`.
- Try to join them with `:riak_core.join('dev1@127.0.0.1')` when in node2's or node3's terminal.
- Use `Riax.put(key, value)` to test it.
- Use `Riax.get(key)` to retrieve the value.
- Use `Riax.ring_status` to see if the handoff is working
